src/protocol/control_layer/ControlMessage.js

```diff
@@ -41,14 +41,47 @@ class ControlMessage {
     }
 
     static registerClass(version, type, clazz) {
+        if (classByType[version] === undefined) {
+            classByType[version] = {}
+        }
         classByType[version][type] = clazz
     }
 
     static getClass(version, type) {
-        if (version !== 0 && version !== 1) {
+        const classesByVersion = classByType[version]
+        if (!classesByVersion) {
             throw new UnsupportedVersionError(version, 'Supported versions: [0, 1]')
         }
+        const clazz = classesByVersion[type]
+        if (!clazz) {
+            throw new UnsupportedTypeError(type, 'Supported types: [1-13]')
```

Comment:
Set of supported types depends on protocol version, and is available in `classesByVersion` as keys. Would be good to construct the error message dynamically to avoid 1) incorrect info (not all types are supported in all versions), and 2) maintenance of this code.

The error message could include the version: `Supported message types in version X: [...]`